File: app/src/main/java/com/example/myanimedata/api/AnimeResponse.java
package com.example.myanimedata.api;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class AnimeResponse {
    @SerializedName("data")
    private final List<AnimeResult> animeResults = new ArrayList<>();

    public List<AnimeResult> getAnimeResults() {
        return animeResults;
    }
}

File: app/src/main/java/com/example/myanimedata/api/AnimeResponse.kt
package com.example.myanimedata.api

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class AnimeResponse (
    @field:SerializedName("data") val animeResults: ArrayList<AnimeResult>? = null
) : Parcelable